libs/navigation/driver/shopify/src/navigation.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { APOLLO_CLIENT_NAME } from '@daffodil/driver/shopify';
+import { DaffNavigationTree } from '@daffodil/navigation';
+import { DaffNavigationServiceInterface } from '@daffodil/navigation/driver';
+
+import {
+  getCollectionMenuQuery,
+  getCollectionQuery,
+} from './query/get-collection-menu';
+import {
+  transformShopifyMenuToNavTree,
+  transformShopifyCollectionToNavItem,
+} from './transforms/navigation-transform';
+
+/**
+ * A service for retrieving navigation data from Shopify collections.
+ *
+ * @inheritdoc
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DaffShopifyNavigationService implements DaffNavigationServiceInterface<DaffNavigationTree> {
+
+  constructor(private apollo: Apollo) {}
+
+  getTree(): Observable<DaffNavigationTree> {
+    return this.apollo.use(APOLLO_CLIENT_NAME)
+      .query({
+        query: getCollectionMenuQuery,
+        variables: {
+          handle: 'collections',
+        },
+      })
+      .pipe(
+        map((result: any) => {
+          const menu = result.data?.menu;
+          return transformShopifyMenuToNavTree(menu);
+        }),
```

Comment:
can you add error handling? or at least a TODO?

Reply:
What kind of handling would you like to see? I probably just want to bubble anything up since we don't have much in the way of well-defined error interfaces for navigation.

Reply:
Todo added. https://github.com/graycoreio/daffodil/pull/4030/commits/b6999fda0bb31ffec6c8b0b25fbde6afa8758054#diff-d5fad62c5f292e095e6a226381bb2e2cd2c3006c228f65327e60cdea8666b036R49